flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/Benchmark.java

```diff
@@ -71,46 +70,59 @@ public static void printHelp() {
 
         StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
         StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
 
-        List<BenchmarkResult> results = new ArrayList<>();
-        for (Map.Entry<String, ?> benchmark : benchmarks.entrySet()) {
-            LOG.info("Running benchmark " + benchmark.getKey() + ".");
-
-            BenchmarkResult result =
-                    BenchmarkUtils.runBenchmark(
-                            tEnv, benchmark.getKey(), (Map<String, ?>) benchmark.getValue());
-
-            results.add(result);
-            LOG.info(BenchmarkUtils.getResultsMapAsJson(result));
+        int index = 0;
+        for (Map.Entry<String, Map<String, Map<String, ?>>> entry : benchmarks.entrySet()) {
+            String benchmarkName = entry.getKey();
+            Map<String, Map<String, ?>> benchmarkMap = entry.getValue();
+
+            LOG.info(
+                    String.format(
+                            "Running benchmark %d/%d: %s",
+                            index++, benchmarks.keySet().size(), benchmarkName));
+
+            try {
+                BenchmarkResult result =
+                        BenchmarkUtils.runBenchmark(tEnv, benchmarkName, benchmarkMap);
+                benchmarkMap.put("results", result.toMap());
+                LOG.info(String.format("Benchmark %s finished.\n%s", benchmarkName, benchmarkMap));
+            } catch (Exception e) {
+                benchmarkMap.put(
+                        "results",
+                        Collections.singletonMap(
+                                "exception",
+                                String.format(
+                                        "%s(%s:%s)",
+                                        e,
+                                        e.getStackTrace()[0].getFileName(),
+                                        e.getStackTrace()[0].getLineNumber())));
+                LOG.info(String.format("Benchmark %s failed.\n%s", benchmarkName, e));
+            }
         }
-
-        String benchmarkResultsJson =
-                BenchmarkUtils.getResultsMapAsJson(results.toArray(new BenchmarkResult[0]));
         System.out.println("Benchmarks execution completed.");
-        System.out.println("Benchmark results summary:");
-        System.out.println(benchmarkResultsJson);
 
+        String benchmarkResultsJson =
+                ReadWriteUtils.OBJECT_MAPPER
+                        .writerWithDefaultPrettyPrinter()
+                        .writeValueAsString(benchmarks);
         if (commandLine.hasOption(OUTPUT_FILE_OPTION.getLongOpt())) {
-            String saveFile = commandLine.getOptionValue(OUTPUT_FILE_OPTION.getLongOpt());
             ReadWriteUtils.saveToFile(saveFile, benchmarkResultsJson, true);
             System.out.println("Benchmark results saved as json in " + saveFile + ".");
+        } else {
+            System.out.println("Benchmark results summary:");
```

Comment:
Would it be simpler and more consistent with the `LOG.info(...)` above by using the following code:

`System.out.println(String.format("Benchmark results summary:\n%s", benchmarkResultsJson))`